sourcegraph/sourcegraph.proto

```diff
@@ -102,6 +102,10 @@ message Repo {
 	// URL. E.g., "github.com/user/repo".
 	string uri = 1 [(gogoproto.customname) = "URI"];
 
+	// Origin is populated for repos fetched via federation or
+	// discovery. It is the hostname of the host that owns the repo.
+	string origin = 21;
```

Comment:
This `origin` field, what is it's relationship to the `clone_url` in `ReposCreateOp`? Do they have different names because of a subtle different meaning, or simply because they were created at different times?

If `mirror` is true, does that mean `origin` is effectively the `clone_url` of the mirror's canonical location?